client/src/entrypoints/snippets/snippet-chooser-modal.js

```diff
@@ -0,0 +1,36 @@
+import $ from 'jquery';
+import { SearchController } from '../../utils/chooserModal';
+
+window.SNIPPET_CHOOSER_MODAL_ONLOAD_HANDLERS = {
```

Comment:
may not be worth changing this approach yet, but I really think we can do this with events.

The core code that inits first will set up an event listener, (e.g. 'wagtail:register-chooser-modal')

Then this code can dispatch an event when initiated such as
```
window.dispatchEvent('wagtail:register-chooser-modal', {detail: {
  choose: ...,
  chosen: ....,
}} );
```

Reply:
For this PR I wanted to stick with simple internal refactoring and not change any APIs. The eventual goal is to make it possible to define a basic chooser with no new JS code at all, and only require a dedicated JS module if the chooser is doing something non-standard. My first attempt at this was to introduce a class-based replacement for ModalWorkflow, where we'd provide a ChooserModalWorkflow subclass and allow further subclassing (ImageChooserModalWorkflow) as required, but the Draftail integration is too closely tied to this "bundle of onload handlers" mechanism and rewriting that was too much of a yak-shave...

My new plan is to keep "bundle of onload handlers" as the outward-facing API, but to implement a factory object that can generate one of those bundles. I'm not clear on what we'd gain from hooking into the browser's event-handling mechanism - as I see it, the functions in the bundle are very much internal to the workings of one specific chooser interface. Having to expose them as a global variable is definitely not my preference, and stashing them in a global 'registry' feels like the same sort of deal.

Reply:
Hey. Thanks for the response. 
I agree we should do this refactor first. 

I really think, longer term, our lightweight framework discussion can shed some light on how to solve this in a consistent way with a class inheritance approach to add behaviour as needed for special cases.